common/datablocks/src/data_block_kernel.rs

```diff
@@ -176,4 +179,49 @@ impl DataBlock {
             }
         }
     }
+
+    pub fn scatter_block(
+        block: &DataBlock,
+        indices: &DataArrayRef,
+        scatter_size: usize
+    ) -> Result<Vec<DataBlock>> {
+        let columns_size = block.num_columns();
+        let mut scattered_columns: Vec<Option<ArrayRef>> = vec![];
+
+        scattered_columns.resize_with(scatter_size * columns_size, || None);
+
+        for column_index in 0..columns_size {
+            let column = block.column(column_index).to_array()?;
```

Comment:
For `constant` column, we can have a faster path

Reply:
Let's optimize it in other patches https://github.com/datafuselabs/datafuse/issues/658